### ISSUE TITLE
proposal: add shorthand rule for throwing mock implementations

### ISSUE THREAD

**Issue by nami8824** [vitest-dev/eslint-plugin-vitest#901]
## Summary

Vitest 4.1 added `mockThrow` and `mockThrowOnce` as shorthand APIs for mocks that throw:

```ts
mock.mockThrow(new Error('error message'))
mock.mockThrowOnce(new Error('first call error'))
```

Currently, `eslint-plugin-vitest` has shorthand rules for returned values and promises:

- `prefer-mock-return-shorthand`
- `prefer-mock-promise-shorthand`

However, throwing mock implementations are still not covered.

This proposes adding a new rule, tentatively named `prefer-mock-throw-shorthand`, that reports `mockImplementation` / `mockImplementationOnce` callbacks whose only behavior is throwing a value.

**Examples**

Invalid:
```ts
vi.fn().mockImplementation(() => {
  throw new Error('error message');
});

mock.mockImplementationOnce(() => {
  throw new Error('error message');
});
```

Valid:
```ts
vi.fn().mockThrow(new Error('error message'));

mock.mockThrowOnce(new Error('error message'));
```
